packages/smooth_app/lib/cache/files/files_cache.dart

```diff
@@ -0,0 +1,37 @@
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:flutter/cupertino.dart';
+import 'package:path/path.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:smooth_app/cache/cache_manager.dart';
+
+part 'files_cache_impl.dart';
+
+/// Files caches work with a [String] key.
+/// This key being the filename.
+typedef FileCache = CacheMap<String, Uint8List>;
+
+/// Cache allowing to store files in the device persistent storage
+class FileCacheManager {
+  const FileCacheManager._();
+
+  static FileCache? _temporaryCache;
+  static FileCache? _longLivingCache;
+
+  static FileCache get temporary {
+    _temporaryCache ??= _ShortLivingFileCacheManagerImpl(
+      subFolderName: 'temp',
```

Comment:
I won't give you an award for the subfolder names: `temp` and `files` :)
What can we do? Anyway you're right, we're better off with subfolders. OK then!

Reply:
I have no perfect idea for these two folders, I have to admit it!
I've renamed the persistent one… to "persistent", but still with no real conviction